src/app/_guards/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, ActivatedRouteSnapshot } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+    constructor(
+        private authService: AuthService,
+        private router: Router) {}
+
+    canActivate(next: ActivatedRouteSnapshot): boolean {
+        const roles = next.data['roles'] as Array<string>;
+        if (roles) {
```

Comment:
This could be simplified quite a bit. combine the logged in check/return and combine the multiple router navigates/access denied areas

Reply:
Agreed, want to be sure whether we want to handle access in the auth guard before I refined it further.